### DISCUSSION TITLE
Import songs with stable's "Date Added" per beatmap and not per set

### DISCUSSION THREAD

**Discussion by EXtremeExploit** [ppy/osu#27641] [Ideas]
Currently lazer when importing maps it will import the whole set with the same added date instead of using each difficulty's date, looking at the code it seems that date added is done per set and not per beatmap, would be nice if it was done per map so the order is exactly the same as stable, this is the ONLY thing thats stopping me from fully switching to lazer
more context here: https://github.com/ppy/osu/issues/13717#issuecomment-1676456150

**Comment by Rekunan** [comment]
I actually kinda dislike having the difficulty's be split apart because of different dates, but I do agree that it would be nice for parity's sake.

**Reply by EXtremeExploit** [reply]
I guess this could be solved when importing maps from stable to have a button to use the stable's date added dates or the current algorithm (be done per set)

**Comment by EXtremeExploit** [comment]
Could this be turned into an issue please? if its what it takes for it to get some kind of priority or visibility for people that would want to tackle it

**Reply by peppy** [reply]
Can you explain the use case here? Generally you'd be importing sets as a.. set. Is this a beatmap modding edge case or something?

**Reply by EXtremeExploit** [reply]
I wouldn't call it a modding edge case but an edge case nonetheless, i just want to keep the exact beatmap order i have on stable over in lazer, i have lots of practice diffs for maps and/or i updated some maps in the past and the update was partial and the set is in 2 places and i got REALLY used to the order of the maps, having all of them on one place would certainly be a bit confusing when im scrolling. Really an edge case

**Comment by Tenexxt** [comment]
Since the Song play (F6 I think?) at the moment sorts songs **only** by the "Date added" sort, it would either need to use the current system while song select having a new one, or you'd have a possibility of listening to the same song for a 5th time because the difficulties were irregularly updated.

Not a good result for 2 in my opinion.

**Reply by EXtremeExploit** [reply]
It could be deduplicated, im asuming this is already done (i really never use song play) since there are multiple maps for the same songs using the exact same mp3, does harumachi clover play 20 times?

**Reply by Tenexxt** [reply]
> since there are multiple maps for the same songs using the exact same mp3, does harumachi clover play 20 times?

As a person who has multiple mapsets of the same song, and downloaded them at the same time - yes, this happens.
This happens even if they are different and the same files (Also kinda of topic but please at least add sorting options for f6)
here is an example of how it looks: 

![image](https://github.com/user-attachments/assets/eb802aa8-af42-4643-80a3-a34593ffe40a)

![image](https://github.com/user-attachments/assets/522783c3-a5da-4ecc-855e-6f6f247a51c7)
